library/core/src/iter/traits/iterator.rs

```diff
@@ -1591,6 +1592,160 @@ pub trait Iterator {
         Flatten::new(self)
     }
 
+    /// Calls the given function `f` for each contiguous window of size `N` over
+    /// `self` and returns an iterator over the outputs of `f`.
+    ///
+    /// In the following example, the closure is called three times with the
+    /// arguments `&['a', 'b']`, `&['b', 'c']` and `&['c', 'd']` respectively.
+    ///
+    /// ```
+    /// #![feature(iter_map_windows)]
+    ///
+    /// let strings = "abcd".chars()
+    ///     .map_windows(|[x, y]| format!("{}+{}", x, y))
+    ///     .collect::<Vec<String>>();
+    ///
+    /// assert_eq!(strings, vec!["a+b", "b+c", "c+d"]);
+    /// ```
+    ///
+    /// Note that the const parameter `N` is usually inferred by the
+    /// destructured argument in the closure.
+    ///
+    /// The returned iterator yields 𝑘 − `N` + 1 items (where 𝑘 is the number of
+    /// items yielded by `self`). If `self` yields fewer than `N` items, the
+    /// iterator returned from this method is empty.
+    ///
+    /// The returned iterator is always fused, because once `self` returns `None`,
+    ///  even if it returns a `Some(T)` again in the next iteration, we cannot put
+    /// it into a contigious array buffer, and thus the returned iterator is also
+    /// fused.
```

Comment:
this is just an explanation of what a fuse is, maybe better to link to `FusedIterator`?